src/com/datacash/training/chapter14/Question16.java

```diff
@@ -0,0 +1,62 @@
+
+package com.datacash.training.chapter14;
+
+interface IntPredicate {
+
+	boolean test( int n );
+}
+
+class MyIntNum {
+
+	private int v;
+
+	public MyIntNum( int x ) {
+		v = x;
+	}
+
+	public int getNum() {
+		return v;
+	}
+
+	// Return true if n is a factor of v.
+	private boolean isFactor( int n ) {
+		return (v % n) == 0;
+	}
+
+	// Return true if n is a factor of i.
+	private boolean isFactor( int n, int i ) {
+		return (i % n) == 0;
+	}
+
+	public boolean hasCommonFactor( int i ) {
+
+		for ( int n = 2 ; n < v / n ; n++ ) {
+			if ( isFactor( n ) && isFactor( n, i ) ) {

```

Comment:
Again, much nicer than the books solution. He doesn't bother to use the predefined isFactor